### PR TITLE
Add cilogon credentials and support

### PR DESCRIPTION
**Pull request by GeorgianaElena**
Reference: https://github.com/2i2c-org/infrastructure/issues/315

### CODE REVIEW

**Comment by GeorgianaElena** [comment]
Friendly ping to @2i2c-org/tech-team :arrow_right:  Do you see any reason for not merging this (given the context in https://github.com/2i2c-org/infrastructure/issues/315#issuecomment-975887945)? (Other than there are no docs around this yet,  but promise to add them once this gets approved)

**Comment by choldgraf** [comment]
Here's the comment where we decided which one to implement: https://github.com/2i2c-org/infrastructure/issues/315#issuecomment-982960242

It looks like the main thing that this does is adds our CILogon credentials to the deployment configuration etc. So is this correct:

1. We have set up an application on CILogon, and this gave us some kind of credential which we're adding in this PR.
2. If a community wants to use CILogon, in their hub configuration we'll need to configure the regex to support it. Here's an example of the configuration we would need:
  - https://github.com/2i2c-org/infrastructure/blob/d3da0f3e722bec6b3a5bc72a5614d0c7f6bbfd08/config/hubs/cloudbank.cluster.yaml#L358-L367
3. So once this PR is merged, we'll be able to add the hub authentication configuration to make Auth0 provide a "CILogon" button for authentication.

Is that right?

**Review by choldgraf — APPROVED** [review]
I think we should merge this one in, and then go through the process of setting up CILogon for a test hub to see if / how it works. I'm +1 to merge as long as we track these next steps in the other issue. Will update it so we don't forget

**Comment by GeorgianaElena** [comment]
I follow @choldgraf's suggestion and merge this, then track follow-up doc PRs in the original issue. Thanks @choldgraf!

File: config/hubs/schema.yaml [hunk 1]
```diff
@@ -179,6 +179,7 @@ properties:
                   - google-oauth2
                   - github
                   - password
+                  - CILogon
```
Comment (damianavila) [inline config/hubs/schema.yaml:182]: I was expecting all lowercase here, but it is not a big deal at all 😉.

Comment (GeorgianaElena) [reply config/hubs/schema.yaml:182]: Ah wow! Yes, lowercase would have been better. Will keep this in mind and when I'll document this, I will also change the casing :sweat_smile: TY!

**Pull request merged**